From 2291bc24c8d932102c1610dbd1ea8947b941f70f Mon Sep 17 00:00:00 2001
From: Sergey Sharybin <sergey@blender.org>
Date: Thu, 27 Mar 2025 17:50:20 +0100
Subject: [PATCH] Skip installation of bvh_build_array.h

Previous fix in this area made it so it consistently installed,
regardless of whether it is a fresh build install, or a subsequent
one. This was the safest as it was done to ensure Blender dependencies
are build in a reproducible manner, as we were reaching the release
phase.

Looking back into this topic: the header is quite big and is only
needed during compilation.
---
 CMakeLists.txt | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b13d2da..a4b116a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -574,8 +574,10 @@ file(GLOB HIPRT_HEADERS "${CMAKE_CURRENT_SOURCE_DIR}/hiprt/*.h")
 install(FILES ${HIPRT_HEADERS}
 		DESTINATION include/hiprt)
 
-# add header files from implementation folder
+# Add header files from implementation folder.
+# Ignore header which is generated by the baking tool and is only needed by the build process.
 file(GLOB HIPRT_IMPL_HEADERS "${CMAKE_CURRENT_SOURCE_DIR}/hiprt/impl/*.h")
+list(REMOVE_ITEM HIPRT_IMPL_HEADERS "${CMAKE_CURRENT_SOURCE_DIR}/hiprt/impl/bvh_build_array.h")
 install(FILES ${HIPRT_IMPL_HEADERS}
 		DESTINATION include/hiprt/impl)
 		
@@ -590,12 +592,6 @@ if(PRECOMPILE)
 			DESTINATION bin)
 endif()
 
-# Baked headers.
-if(BAKE_COMPILED_KERNEL)
-	install(FILES ${KERNEL_HIPRT_H}
-			DESTINATION include/hiprt/impl)
-endif()
-
 # Project: Unit Test
 if(NOT NO_UNITTEST)
 
-- 
2.48.1

